### PR TITLE
Feature/new examples

### PR DESCRIPTION
**Pull request by JesStruk**
# Description

The module for provisioning vm's/dc's was extended by the basic and advanced examples to cover all the functionality of the module. Also when not provided with dns server ip's which in the standard should be the on premise dc's, by default the nic will take the 168.63.129.16 IP for the Azure DNS.

# Change overview (tick true):

- [ ] Changed minor things, so it is deployable via terraform
- [ ] Fix minor bugs
- [ ] Added examples

# Version information: 

- Previous Version: `0.0.0`

- Next Version based on [Semantic Versioning](https://semver.org/#summary) (see above): `1.0.0`

# How Has This Been Tested?

- [ ] Terraform Plan
- [ ] Tested by deployment via terraform apply

# Checklist:

- [ ] I have run tests and documented them above
- [ ] I have performed a self-review of my own code
- [ ] I have updated the documentation

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -1,3 +1,15 @@
 locals {
-  
-}
\ No newline at end of file
+  public_ip = {
+      name = coalesce(var.name_overrides.public_ip, "pip-vm-${var.virtual_machine_config.hostname}") # change to naming convention= 
+  }
+
+  nic = {
+      name = coalesce(var.name_overrides.nic, replace(replace("nic-${var.virtual_machine_config.hostname}-${var.subnet.address_prefixes[0]}", ".", "-"), "/", "-")) # replace regex
```
Comment (TheodorWulff) [inline locals.tf:7]: use regex

File: terraform.tf [hunk 2]
```diff
@@ -6,3 +6,7 @@ terraform {
     }
   }
 }
+
+provider "azurerm" {
```
Comment (TheodorWulff) [inline terraform.tf:10]: remove this from the module. This should only belong in the code that calls this module.

**Review by TheodorWulff — CHANGES_REQUESTED** [review]
Added some comments

File: examples/basic/main.tf [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+module "virtual_machine" {
+    source = "../.."
+    virtual_machine_config = {
+        hostname = "CUSTAPP001"
+        location = local.location
+        admin_username = "local_admin"
+        size = "Standard_D2_v5"
+        os_sku = "2022-Datacenter"
+        os_version = "latest"
```
Comment (QBY-ChristianHartmann) [inline examples/basic/main.tf:13]: is optional, leave out of basic example

File: locals.tf (hunk 1 above)
Comment (JesStruk) [reply locals.tf]: replace uses now regex "/[./]/" - no double replace function

File: examples/basic/main.tf (hunk 3 above)
Comment (QBY-ChristianHartmann) [inline examples/basic/main.tf:13]: is optional, leave out of example

**Review by TheodorWulff — CHANGES_REQUESTED** [review]
Minor bug in naming

**Review by TheodorWulff — DISMISSED** [review]
Looks good now

**Comment by QBY-ThimoLimpert** [comment]
Please change the readme, add the second example and provide a little bit more info then `easy to use `:D

**Review by QBY-ThimoLimpert — APPROVED** [review]

**Review by QBY-ChristianHartmann — APPROVED** [review]

**Pull request merged**